### PR TITLE
Geowave-862: HBase distributed filters and coprocessors

### PR DESCRIPTION
**Pull request by blastarr**

### CODE REVIEW

File: extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/AggregationEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+			if (request.getAggregation().hasParams()) {
+				byte[] parameterBytes = request.getAggregation().getParams().toByteArray();
+				final Persistable aggregationParams = PersistenceUtils.fromBinary(
+						parameterBytes,
+						Persistable.class);
+				aggregation.setParameters(aggregationParams);
+			}
+		}
+		catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+			System.err.println(e.getMessage());
```
Comment (rfecher) [inline extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/AggregationEndpoint.java]: Log exceptions

File: extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/AggregationEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+						filterList.addFilter(rangeFilter);
+					}
+				}
+				catch (Exception e) {
+					System.err.println(e.getMessage());
+					e.printStackTrace();
+				}
+			}
+			else {
+				System.err.println("Input range filter is undefined. ");
```
Comment (rfecher) [inline extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/AggregationEndpoint.java]: Log

File: extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/HBaseConstraintsQuery.java [hunk 3]
```diff
@@ -163,6 +191,177 @@ protected boolean isAggregation() {
...
+
+		// Create the multi-range filter
+		try {
+			MultiRowRangeFilter filter = new MultiRowRangeFilter(
+					rowRanges);
+
+			return filter;
+		}
+		catch (IOException e) {
+			e.printStackTrace();
```
Comment (rfecher) [inline extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/HBaseConstraintsQuery.java]: ...

File: extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/HBaseDistributableFilter.java [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+				getAdapterEncoding(dataAdapter),
+				new PrimaryIndex(
+						null,
+						model));
+	}
+
+	protected boolean applyRowFilter(
+			final CommonIndexedPersistenceEncoding encoding ) {
+		if (filterList == null) {
+			System.err.println("FILTER IS NULL");
```
Comment (rfecher) [inline extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/HBaseDistributableFilter.java]: log

File: extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/HBaseDistributableFilter.java [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+		}
+
+		filterList.clear();
+
+		for (Persistable decodedFilter : decodedFilterList) {
+			if (decodedFilter instanceof DistributableQueryFilter) {
+				filterList.add((DistributableQueryFilter) decodedFilter);
+			}
+			else {
+				System.out.println("Unrecognized type for decoded filter!" + decodedFilter.getClass().getName());
```
Comment (rfecher) [inline extensions/datastores/hbase/src/main/java/mil/nga/giat/geowave/datastore/hbase/query/HBaseDistributableFilter.java]: log

**Comment by blastarr** [comment]
Changed all the system print and exception stack trace calls to use the logger

File: core/store/src/main/java/mil/nga/giat/geowave/core/store/query/aggregate/CountAggregation.java [hunk 6]
```diff
@@ -42,6 +42,6 @@ public void setParameters(
 
 	@Override
 	public void clearResult() {
-		countResult = new CountResult();
```
Comment (rfecher) [inline core/store/src/main/java/mil/nga/giat/geowave/core/store/query/aggregate/CountAggregation.java]: why this change?
